src/sudoku.py

```diff
@@ -12,6 +13,21 @@ def __init__(self, grid):
         for i, j in product((0, 3, 6), (0, 3, 6)):
             self.__squares.append([self.__rows[x][y] for x in range(i, i+3) for y in range(j, j+3)])
 
+    def __exclude_equal_alternatives__(self, grid_view):
+        is_any_cell_solved = False  # флаг что хоть одна клетка решена
+        for i in range(9):
+            alternatives = Counter([grid_view[i][j].alternatives for j in range(9)])  # формируем словарь алтернативы в пачке(строка, столбец или квадрант):количество таких альтернатив в пачке 
+            exclude_set = set()
+            for alternative in alternatives:
```

Comment:
А вот здесь уже `alternatives` 
```suggestion
            for alternatives in twin_alternatives_counter:
```